src/query/query_parser/query_parser.rs

```diff
@@ -456,12 +468,12 @@ fn convert_literal_to_query(logical_literal: LogicalLiteral) -> Box<Query> {
     }
 }
 
-fn convert_to_query(logical_ast: LogicalAST) -> Box<Query> {
+fn convert_to_query(logical_ast: LogicalAST, should_match_all: bool) -> Box<Query> {
     match trim_ast(logical_ast) {
         Some(LogicalAST::Clause(trimmed_clause)) => {
             let occur_subqueries = trimmed_clause
                 .into_iter()
-                .map(|(occur, subquery)| (occur, convert_to_query(subquery)))
+                .map(|(occur, subquery)| (occur, convert_to_query(subquery, should_match_all)))
```

Comment:
I don't think this is correct to recursively apply this value.

What happens if we have queries like :
`A AND NOT <some_stopword>`

This "what happens if empty logic should on" if the entire query is empty

Reply:
Ok, I'll try to find a more correct way.

Reply:
To be honest I am not entirely sure of what the best definition should be. You just write a bunch of unit test beforehands, and we can discuss the best outcome when `should_match_all` is `true` or `false`.

- `something -<a stop word>`
- `something something_else`
- `something AND something_else`
- `something AND NOT something_else`
- `something OR something_else`

Reply:
So, I wrote this at the moment:

```rust
pub fn test_logic_behind_should_match_all() {
    let query_parser = make_query_parser();

    // something -<a stop word>
    assert!(query_parser.parse_query("toto -such").is_ok());

    // something something_else
    assert!(query_parser.parse_query("toto teta").is_ok());

    // something AND something_else
    assert!(query_parser.parse_query("toto AND foo").is_ok());

    // something AND NOT something_else
    assert!(query_parser.parse_query("toto AND NOT bar").is_ok());

    // something OR something_else
    assert!(query_parser.parse_query("toto OR foobar").is_ok());
}
```

I plan to make the `query_parser` variable `mut` and toggle the `should_match_all` on/off on need. It's ok, or should I make a new `query_parser` from scratch for every test case? The queries are correct or did you expect something different? Sorry for not reaching out before, I got kind of busy :)

Reply:
@druzn3k 

Sorry for the 3month iatus. Of course all of these queries should pass.
My concern is more about what they should return...

For instance, if empty match all,  `term_that_exists -term_that_does_not_exist`  will return no document, which is rather unexpected.

I think the match_all_if_query should not apply if this is part of another query.